admin/resources/js/components/poolCandidate/UpdatePoolCandidate.tsx

```diff
@@ -0,0 +1,313 @@
+} from "../../api/generated";
+import errorMessages from "../form/errorMessages";
+import Submit from "../form/Submit";
+import Select from "../form/Select";
+import Input from "../form/Input";
+import { enumToOptions, getId, notEmpty } from "../../helpers/util";
+import MultiSelect from "../form/MultiSelect";
+import { unpackIds, unpackMaybes } from "../form/formUtils";
+
+type Option<V> = { value: V; label: string };
+
+type FormValues = Pick<
+  PoolCandidate,
+  | "cmoIdentifier"
+  | "expiryDate"
+  | "hasDiploma"
+  | "hasDisability"
+  | "isIndigenous"
+  | "isVisibleMinority"
+  | "isWoman"
+  | "languageAbility"
+  | "expectedSalary"
+  | "locationPreferences"
+  | "status"
+> & {
+  acceptedOperationalRequirements: string[] | undefined;
+  cmoAssets: string[] | undefined;
+  expectedClassifications: string[] | undefined;
+};
+
+interface UpdatePoolCandidateProps {
+  classifications: Classification[];
+  cmoAssets: CmoAsset[];
+  initialPoolCandidate: PoolCandidate;
+  locale: "en" | "fr";
+  operationalRequirements: OperationalRequirement[];
+  handleUpdatePoolCandidate: (
+    id: string,
+    data: UpdatePoolCandidateInput,
+  ) => Promise<UpdatePoolCandidateMutation["updatePoolCandidate"]>;
+}
+
+export const UpdatePoolCandidateForm: React.FunctionComponent<UpdatePoolCandidateProps> =
+  ({
+    classifications,
+    cmoAssets,
+    initialPoolCandidate,
+    locale,
+    operationalRequirements,
+    handleUpdatePoolCandidate,
+  }) => {
+    const dataToFormValues = (
+      data: PoolCandidate | UpdatePoolCandidateMutation["updatePoolCandidate"],
+    ): FormValues => ({
+      // TODO: Convert rest of data to form values format (specifically the multi select).
+      ...data,
+      acceptedOperationalRequirements: unpackIds(
+        data?.acceptedOperationalRequirements,
+      ),
+      cmoAssets: unpackIds(data?.cmoAssets),
+      expectedClassifications: unpackIds(data?.expectedClassifications),
+      expectedSalary: unpackMaybes(data?.expectedSalary),
+      locationPreferences: unpackMaybes(data?.locationPreferences),
+    });
+
+    const formValuesToSubmitData = (
+      values: FormValues,
+    ): UpdatePoolCandidateInput => ({
+      ...values,
+      acceptedOperationalRequirements: {
+        sync: values.acceptedOperationalRequirements,
+      },
+      expectedClassifications: {
+        sync: values.expectedClassifications,
+      },
+      cmoAssets: {
+        sync: values.cmoAssets,
+      },
+    });
+
+    const methods = useForm<FormValues>({
+      defaultValues: dataToFormValues(initialPoolCandidate),
+    });
+    const { handleSubmit, reset } = methods;
+
+    const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
+      await handleUpdatePoolCandidate(
+        initialPoolCandidate.id,
+        formValuesToSubmitData(data),
+      )
+        .then((resolved) => reset(dataToFormValues(resolved))) // Reset form with returned data. This resets isDirty flag.
```

Comment:
Right now in storybook the entire form goes blank after submitting - not sure why, but that's not supposed to happen :/

Reply:
It was because I set the return value to null in the story haha. It's fixed now.